File: 131-PalindromePartitioning/131-PalindromePartitioning.java
// Last updated: 8/1/2025, 1:57:40 PM
class Solution {
    public List<List<String>> partition(String s) {
        List<List<String>> ans = new ArrayList<>();
		List<String> ll = new ArrayList<>();
		Partitioning(s, ll, ans);
		return ans;
    }
    public static void Partitioning(String ques, List<String> ll,           List<List<String>> ans) {
		if (ques.length() == 0) {
//			System.out.println(ll);
			ans.add(new ArrayList<String>(ll));
			return;
		}
		for (int cut = 1; cut <= ques.length(); cut++) {
			String s = ques.substring(0, cut);
			if (isPalindromic(s)) {
				ll.add(s);
				Partitioning(ques.substring(cut), ll, ans);
				ll.remove(ll.size() - 1);
			}
		}
	}
	public static boolean isPalindromic(String s) {
		int i = 0;
		int j = s.length() - 1;
		while (i < j) {
			if (s.charAt(i) != s.charAt(j)) {
				return false;
			}
			i++;
			j--;
		}
		return true;
	}

}

File: 131-PalindromePartitioning/131-PalindromePartitioning.py
# Last updated: 1/8/2026, 7:43:07 PM
1class Solution:
2    def partition(self, s: str) -> List[List[str]]:
3        temp=[]
4        lst=[]
5
6        self.part(s,temp,lst)
7        return lst
8
9    def part(self,s:str,temp:List[str],lst:List[List[str]]):
10
11        if len(s)==0:
12            lst.append(temp.copy())
13            return
14        
15        for cut in range(1,len(s)+1):
16            st=s[:cut]
17            if self.ispalind(st):
18                temp.append(st)
19                self.part(s[cut:],temp,lst)
20                temp.pop()
21    
22    def ispalind(self,st:str)->bool:
23        return st== st[::-1]
24        
25